include/boost/geometry/formulas/distance_point_segment.hpp

```diff
@@ -0,0 +1,335 @@
+// Boost.Geometry
+
+// Copyright (c) 2016-2017 Oracle and/or its affiliates.
+
+// Contributed and/or modified by Vissarion Fysikopoulos, on behalf of Oracle
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_FORMULAS_CROSS_TRACK_GEO_HPP
+#define BOOST_GEOMETRY_FORMULAS_CROSS_TRACK_GEO_HPP
+
+#include <boost/geometry/formulas/result_direct.hpp>
+
+#ifndef BOOST_GEOMETRY_DETAIL_POINT_SEGMENT_DISTANCE_MAX_STEPS
+#define BOOST_GEOMETRY_DETAIL_POINT_SEGMENT_DISTANCE_MAX_STEPS 100
+#endif
+
+/*!
+\brief Algorithm to compute the distance between a segment and a point using
+       direct and inverse geodesic problems as subroutines. The algorithm
+       approximates the distance by an iterative Newton method.
+\see C.F.F.Karney - Geodesics on an ellipsoid of revolution,
+      https://arxiv.org/abs/1102.1215
+*/
+
+namespace boost { namespace geometry { namespace formula
+{
+
+template
+<
+        typename CT,
+        typename Units,
+        typename Inverse_type_azimuth,
+        typename Inverse_type_distance,
+        typename Direct_type,
+        bool EnableClosestPoint = false
+>
+class distance_point_segment{
+
+public:
+
+    struct result_distance_point_segment
+    {
+        result_distance_point_segment()
+            : distance(0)
+            , closest_point_lon(0)
+            , closest_point_lat(0)
+        {}
+
+        CT distance;
+        CT closest_point_lon;
+        CT closest_point_lat;
+    };
+
+    template <typename Spheroid>
+    result_distance_point_segment
+    static inline non_iterative_case(CT lon1, CT lat1, //p1
+                                     CT lon2, CT lat2, //p2
+                                     Spheroid const& spheroid)
+    {
+        result_distance_point_segment result;
+        result.distance = Inverse_type_distance::apply(lon1, lat1,
+                                                       lon2, lat2,
+                                                       spheroid).distance;
+        if (EnableClosestPoint)
+        {
+            result.closest_point_lon = lon1;
+            result.closest_point_lat = lat1;
+        }
+        return result;
+    }
+
+    template <typename Spheroid>
+    result_distance_point_segment
+    static inline apply(CT lon1, CT lat1, //p1
+                        CT lon2, CT lat2, //p2
+                        CT lon3, CT lat3, //query point p3
+                        Spheroid const& spheroid,
+                        CT earth_radius =
+                           geometry::srs::sphere<CT>().get_radius<1>())
```

Comment:
What's the point of passing WGS84 mean radius here? AFAIU it's redundant and forces the user to pass correct radius, consistent with spheroid. I guess it's for calculating some initial values before iteration. But shouldn't it simply be calculated from passed spheroid internally?